### PR TITLE
inits esling config + prettier

### LINKED ISSUE #4: Add eslint config & make sure prettier is working
Lets make the eslint rules as strict as possible so that when other people work on the project we are all contributing code in the same format

### PR DESCRIPTION
**Pull request by LDNCWRB**
### What changes have you made?

- Installed eslint + eslint.json updates

### Which issue(s) does this PR fix?

- Fixes #4 

### How to test 

- Format on save

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  "rules": {
+    "strict": "error",
+    "indent": ["error", 2],
+    "linebreak-style": ["error", "unix"],
+    "quotes": ["error", "double"],
+    "semi": ["error", "always"],
+    // NextJs specific fix: suppress errors for missing 'import React' in files for nextjs
+    "react/react-in-jsx-scope": "off",
+    // NextJs specific fix: allow jsx syntax in js files
+    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }], //should add ".ts" if typescript project
```
Comment (Jbarget) [inline .eslintrc.json]: we need to do what the comment says and add `.ts` and `.tsx` into the array

**Review by Jbarget — CHANGES_REQUESTED** [review]
can we add `eslint-plugin-jsx-a11y` too and use it like below

image

**Comment by Jbarget** [comment]
can we delete the `package-lock.json` file too :)

**Comment by Jbarget** [comment]
when there are merge conflicts with lock files **never** go in and manually sort the conflicts out, just delete the lock file (`yarn.lock` or `package-lock.json`) and run `yarn` if you're using yarn or `npm i` if you're using npm

so in this case:
sort out the package.json conflicts, then follow the steps above

**Comment by Jbarget** [comment]
do eslint/typescript errors show as expected for you on this branch? I checked last night and it wasn't for me

**Comment by raphaelj274** [comment]
> do eslint/typescript errors show as expected for you on this branch? I checked last night and it wasn't for me

they don't because ts and eslint were not installed on this branch, now that setup is merged we will pull and rebase to check if its working

**Comment by raphaelj274** [comment]
sorry, need to look at the package json again

**Comment by raphaelj274** [comment]
> do eslint/typescript errors show as expected for you on this branch? I checked last night and it wasn't for me

typescript errors working but were not sure how to check for eslint errors as surely prettier is getting rid of them on save?

**Comment by Jbarget** [comment]
if you make an eslint error but dont save then you should still see the error

**Comment by Jbarget** [comment]
when this is ready for re-review click the circular arrows next to my name in the reviewers column, thats how you would let me know when to review again

image

**Comment by raphaelj274** [comment]
hey we're a bit stuck on setting up eslint, we're gonna try create the form in the meantime, but when you have a chance could we go over this?

**Comment by Jbarget** [comment]
this PR can be closed and the branch deleted once #19 goes in

**Pull request closed without merging**